narwhal/network/src/failpoints.rs

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+use anemo_tower::callback::{MakeCallbackHandler, ResponseHandler};
+use fail::fail_point;
+
+#[derive(Clone, Default)]
+pub struct FailpointsMakeCallbackHandler {}
+
+impl FailpointsMakeCallbackHandler {
+    pub fn new() -> Self {
+        Self {}
+    }
+}
+
+impl MakeCallbackHandler for FailpointsMakeCallbackHandler {
+    type Handler = FailpointsResponseHandler;
+
+    fn make_handler(&self, _request: &anemo::Request<bytes::Bytes>) -> Self::Handler {
+        FailpointsResponseHandler {}
+    }
+}
+
+pub struct FailpointsResponseHandler {}
+
+impl ResponseHandler for FailpointsResponseHandler {
```

Comment:
Btw, not sure if you planned it, but it would be good to have delays both on request and response side(as those surface different kinds of networking issues).

Perhaps rename this failpoint to rpc-response-delay then?